src/main/resources/avro/readmethods.avdl

```diff
@@ -14,8 +14,10 @@ to `ReferenceSet`s containing that same `Reference`. If no reference is
 specified, all `ReadGroup`s must be aligned to the same `ReferenceSet`.
 */
 record SearchReadsRequest {
-  /** If nonempty, restrict this query to reads within the given readgroups. */
-  array<string> readGroupIds = [];
+  /**
+  The ReadGroup to search.
+  */
+  string readGroupId;
```

Comment:
Ah, I see, now I understand; thanks for your patience @calbach and @lh3. The reads we return are ordered by position, so it's useful for the client to be able to get all of the reads across multiple variant sets back sorted by genomic position.

My point about overhead was more from the perspective of the client/server exchanges @calback --- would there be substantially more protocol chat if we ask the client to make these requests sequentially? I would say the answer is no, assuming the number of records  returned *r* is larger than the page size *p*.

On page size, I think we have to assume that for anything but trivial queries, that *r* >> *p*, and that the client must always assume that there will be paging. Any client library worthy of the name will abstract away the details of paging. For example, in the reference implementation we use iterators for all protocol queries, and paging happens transparently behind the scenes. I think we should assume that there will be a client library, and that people writing real applications will not be hand-crafting JSON documents and dealing with pagination.

Which brings us back to multi-BAM retrieval. Why should we complicate the server and the protocol, when this can _easily_ be achieved by the client? For a *k*-way merge, all I have to do is make *k* iterators (one for each ReadGroupSet) and merge the reads on the fly. The end-user will simply see the required stream of records, in sorted order. 

There is a fundamental tactical choice here: do we (1) build a protocol that is a set of well-defined simple operations that are composable on the client side to give arbitrarily complex behaviour; or (2) do we try to build every conceivable operation that a client might require into the protocol? I think (1) is  preferable, for a multitude of reasons. 

My argument here is of course very extreme: multi-merge is an important piece of functionality. However, I do think it belongs on the client side. I also think that anything that _can_ go on the client side should be left there, until there is a compelling performance case for moving it into the server.

Apologies for such a long reply, but I do think that this is an important point, and feeds into several of the discussions that are happening at the moment.